ts/src/bitmex.ts

```diff
@@ -2109,15 +2198,21 @@ export default class bitmex extends Exchange {
         const datetime = this.safeString (position, 'timestamp');
         const crossMargin = this.safeValue (position, 'crossMargin');
         const marginMode = (crossMargin === true) ? 'cross' : 'isolated';
-        let notional = undefined;
-        if (market['quote'] === 'USDT' || market['quote'] === 'USD' || market['quote'] === 'EUR') {
-            notional = Precise.stringMul (this.safeString (position, 'foreignNotional'), '-1');
-        } else {
-            notional = this.safeString (position, 'homeNotional');
+        const notionalString = Precise.stringAbs (this.safeString (position, 'foreignNotional', 'homeNotional'));
+        const settleCurrencyCode = this.safeString (market, 'settle');
+        const maintenanceMargin = this.convertToRealAmount (settleCurrencyCode, this.safeString (position, 'maintMargin'));
+        const unrealisedPnl = this.convertToRealAmount (settleCurrencyCode, this.safeString (position, 'unrealisedPnl'));
+        const contracts = this.parseNumber (Precise.stringAbs (this.safeString (position, 'currentQty')));
+        const contractSize = this.safeNumber (market, 'contractSize');
+        let side = undefined;
+        const homeNotional = this.safeString (position, 'homeNotional');
+        if (homeNotional !== undefined) {
+            if (homeNotional[0] === '-') {
```

Comment:
@carlosmiei u sure this is a good way? maybe not quite expressive and also other langs would break on that. can we add `this.isNegativeString ()` in base ?

Reply:
asking this because it is used in different places of the lib

Reply:
i haven't faced similar before, so seemed strange to me,
I thought not all langs would accept index on string, but our current langs handle it, so it's ok.